src/main/java/org/sunbird/profile/service/ProfileServiceImpl.java

```diff
@@ -771,7 +771,7 @@ public SBApiResponse bulkUpload(MultipartFile mFile, String orgId, String channe
 			response.getResult().putAll(uploadedFile);
 			uploadedFile.put(Constants.ORG_NAME, channel);
 			uploadedFile.put(Constants.X_AUTH_TOKEN, userAuthToken);
-			kafkaProducer.push(serverConfig.getUserBulkUploadTopic(), uploadedFile);
+			kafkaProducer.pushWithKey(serverConfig.getUserBulkUploadTopic(), uploadedFile, orgId);
```

Comment:
What is the use of this key -- if it is not utilised at the consumption ?
The idea should be:
use the key value at the consumption... 
push the file with specific key value --> ex -> bulkUploadFile
also push the user create request to in the kafka with key --> ex -> userCreateReq

At the consumer -- check the key value and process the record accordingly...

Reply:
Using CompletableFuture.runAsync() in your consumer means 
that message processing can happen asynchronously and potentially 
concurrently across multiple threads. While messages within the same 
partition and with the same key will be processed sequentially, messages
from different partitions or with different keys may be processed 
concurrently.